calicoctl/commands/resources.go

```diff
@@ -178,25 +183,51 @@ func executeConfigCommand(args map[string]interface{}, action action) commandRes
 		count[kind] = count[kind] + 1
 		results.numResources = results.numResources + 1
 	}
-	if len(count) == 1 {
+	if len(count) == 1 || singleKind {
 		results.singleKind = kind
 	}
 
 	// Now execute the command on each resource in order, exiting as soon as we hit an
 	// error.
 	export := argutils.ArgBoolOrFalse(args, "--export")
-	nameSpecified := argutils.ArgStringOrBlank(args, "<NAME>")
+	nameSpecified := false
+	emptyName := false
+	switch a := args["<NAME>"].(type) {
+	case string:
+		nameSpecified = len(a) > 0
+		_, ok := args["<NAME>"]
+		emptyName = !ok || !nameSpecified
```

Comment:
isn't `nameSpecified` always going to be `false` at this point, since it's initialized to false right above?